doc/source/blueprint_cce_vault.rst

```diff
@@ -0,0 +1,592 @@
+       template {
+         destination = "/tls/ca.pem"
+         contents = <<EOT
+       {{- with secret "/secret/data/tls/ca" }}{{ .Data.data.certificate }}{{ end }}
+       EOT
+       }
+
+       template {
+         destination = "/tls/server/server.pem"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_server_cert_path }}" }}{{ .Data.data.certificate }}
+       {{ .Data.data.private_key }}{{ end }}
+       EOT
+       }
+       template {
+         destination = "/tls/server/tls.crt"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_server_cert_path }}" }}{{ .Data.data.certificate }}{{ end }}
+       EOT
+       }
+       template {
+         destination = "/tls/server/tls.key"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_server_cert_path }}" }}{{ .Data.data.private_key }}{{ end }}
+       EOT
+       }
+
+       template {
+         destination = "/tls/client/client.pem"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_client_cert_path }}" }}{{ .Data.data.certificate }}
+       {{ .Data.data.private_key }}{{ end }}
+       EOT
+       }
+       template {
+         destination = "/tls/client/tls.crt"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_client_cert_path }}" }}{{ .Data.data.certificate }}{{ end }}
+       EOT
+       }
+       template {
+         destination = "/tls/client/tls.key"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_client_cert_path }}" }}{{ .Data.data.private_key }}{{ end }}
+       EOT
+       }
+
+.. code-block:: bash
+
+   $ kubectl apply -f zookeeper-cm.yaml
+
+5. Create Zookeeper Headless service to be used by pod to build quorum
+
+.. code-block:: yaml
+   :caption: zookeeper-svc.yaml
+
+    ---
+    name: "zookeeper-svc"
+    namespace: "zookeeper"
+    apiVersion: v1
+    kind: Service
+    spec:
+      # Not exposing in the cluster
+      clusterIP: None
+      # Important to start up
+      publishNotReadyAddresses: true
+      selector:
+        app: zookeeper
+      ports:
+      - port: 2281
+        name: client
+        targetPort: client
+        protocol: TCP
+      - port: 2888
+        name: server
+        targetPort: server
+        protocol: TCP
+      - port: 3888
+        name: election
+        targetPort: election
+        protocol: TCP
+
+.. code-block:: bash
+
+   $ kubectl apply -f zookeeper-svc.yaml
+
+6. Create Frontend service
+
+.. code-block:: yaml
+   :caption: zookeeper-svc-public.yaml
+
+   apiVersion: v1
+   kind: Service
+   spec:
+     clusterIP: None
+     ports:
+     - name: client
+       port: 2281
+       protocol: TCP
+       targetPort: client
+     selector:
+       app: zookeeper
+     sessionAffinity: None
+     type: ClusterIP
+
+.. code-block:: bash
+
+   $ kubectl apply -f zookeeper-svc-public.yaml
+
+7. Create StatefulSet replacing `<VAULT_PUBLIC_ADDR>` with address of available
```

Comment:
with the address of the Valut server [or with the addresses of the available Vault servers]